pkg/controller/servicebinding/retriever.go

```diff
@@ -32,27 +33,46 @@ func createServiceIndexPath(name string, gvk schema.GroupVersionKind) []string {
 
 }
 
-func buildServiceEnvVars(svcCtx *serviceContext, globalNamePrefix string) (map[string]string, error) {
-	prefixes := []string{}
-	if len(globalNamePrefix) > 0 {
-		prefixes = append(prefixes, globalNamePrefix)
+func buildServiceEnvVars(svcCtx *serviceContext, globalNameStrategy string) (map[string]string, error) {
+	if globalNameStrategy != "" && svcCtx.namingStrategy == nil {
+		svcCtx.namingStrategy = &globalNameStrategy
 	}
-	if svcCtx.namePrefix != nil && len(*svcCtx.namePrefix) > 0 {
-		prefixes = append(prefixes, *svcCtx.namePrefix)
+
+	namingTemplate := ""
+	prefixes := []string{}
+
+	listEnvVars, err := envvars.Build(svcCtx.envVars, prefixes...)
+	if err != nil {
+		return nil, err
 	}
-	if svcCtx.namePrefix == nil {
-		prefixes = append(prefixes, svcCtx.service.GroupVersionKind().Kind)
+
+	if svcCtx.bindAsFiles && svcCtx.namingStrategy == nil {
+		namingTemplate = naming.Strategies["bindAsFiles"]
+	} else if svcCtx.namingStrategy == nil {
+		namingTemplate = naming.Strategies["default"]
+	} else if svcCtx.namingStrategy != nil {
+		if v, ok := naming.Strategies[*svcCtx.namingStrategy]; ok {
+			namingTemplate = v
+		} else {
+			namingTemplate = *svcCtx.namingStrategy
+		}
 	}
 
-	return envvars.Build(svcCtx.envVars, prefixes...)
+	svcCtx.namingStrategy = &namingTemplate
```

Comment:
It's misleading to name the field `namingStrategy` when its value is the template.

Reply:
updated to `namingTemplate`